Prep for v1.0.0 release (#325)
bottlerocket-os/bottlerocket-update-operator: README.md

The controller orchestrates updates across your cluster, while the agent is responsible for periodically querying for Bottlerocket updates, draining the node, and performing the update when asked by the controller.
The agent performs all cluster object mutation operations via the API Server, which performs additional authorization using the Kubernetes TokenReview API -- ensuring that any request associated with a node is being made by the agent pod running on that node.
Further, `cert-manager` is required in order for the API server to use a CA certificate to communicate over SSL with the agents.
Updates to Bottlerocket are rolled out in [waves](https://github.com/bottlerocket-os/bottlerocket/tree/develop/sources/updater/waves) to reduce the impact of issues; the nodes in your cluster may not all see updates at the same time.

## Getting Started

### Installation

1. Brupop uses [cert-manager](https://cert-manager.io) to manage self-signed certificates used by the Bottlerocket update operator. To install cert-manager:

```sh
kubectl apply -f \
  https://github.com/cert-manager/cert-manager/releases/download/v1.8.2/cert-manager.yaml
```

> 2. The Bottlerocket update operator can then be installed using the recommended configuration defined in [bottlerocket-update-operator.yaml](https://github.com/bottlerocket-os/bottlerocket-update-operator/blob/0.2.2/yamlgen/deploy/bottlerocket-update-operator.yaml).

Reviewer: Shouldn't this be updated to 1.0.0?

Author: Good catch!! Thank you! 👏🏼

This YAML file can also be found in the [Brupop release artifacts](https://github.com/bottlerocket-os/bottlerocket-update-operator/releases):

```sh
kubectl apply -f ./bottlerocket-update-operator.yaml
```

This will create the required namespace, custom resource definition, roles, deployments, etc., and use the latest update operator image available in [Amazon ECR Public](https://gallery.ecr.aws/bottlerocket/bottlerocket-update-operator).

> Note: The above link points to the latest version of the Update Operator, `v0.2.2`.

Reviewer 2: ```suggestion
Note: The above link points to the latest version of the Update Operator, `v1.0.0`.
```

Be sure to use the git tag for the Update Operator release that you plan to deploy.

> 3. Label nodes with `bottlerocket.aws/updater-ointerface-version=2.0.0` to indicate they should be automatically updated.

Reviewer 2: I think this is a typo, not sure:

```suggestion
3. Label nodes with `bottlerocket.aws/updater-interface-version=2.0.0` to indicate they should be automatically updated.
```

              value: "1"
```

#### Set an Update Time Window
`UPDATE_WINDOW_START` and `UPDATE_WINDOW_STOP` can be used to specify the time window in which updates are permitted.
When `UPDATE_WINDOW_START` and `UPDATE_WINDOW_STOP` are both 0:0:0 (default), the feature is disabled.

> To enable this feature, go to `bottlerocket-update-operator.yaml`, change `UPDATE_WINDOW_START` and `UPDATE_WINDOW__STOP` to a `hour:minute:second` formate value (UTC (24-hour time notation)). Note that `UPDATE_WINDOW_START` is inclusive and `UPDATE_WINDOW_STOP` is exclusive.

Reviewer 2: Pre-existing typo.

```suggestion
To enable this feature, go to `bottlerocket-update-operator.yaml`, change `UPDATE_WINDOW_START` and `UPDATE_WINDOW__STOP` to a `hour:minute:second` formatted value (UTC (24-hour time notation)). Note that `UPDATE_WINDOW_START` is inclusive and `UPDATE_WINDOW_STOP` is exclusive.
```

kubectl label node $(kubectl get nodes -o jsonpath='{.items[*].metadata.name}') bottlerocket.aws/updater-interface-version=2.0.0
```

### A Note About Removing Labels

> (TODO - is this still true?)

Author: cc @gthao313 - You had done alot of work in this release for cleanup. Is this chunk still relevant?

Reviewer 3: > you must manually clean up the Custom Resource if you choose to delete only the `updater-interface-version`. The Custom Resource can be deleted like so:

this's out of date. Brupop will automatically garbage collect the Custom Resource `brs`. Therefore, I think we just need remove this that sentence.

  - `328549459982.dkr.ecr.ap-northeast-1.amazonaws.com`
  - `328549459982.dkr.ecr.ap-northeast-2.amazonaws.com`
  - `328549459982.dkr.ecr.ap-south-1.amazonaws.com`
  - `328549459982.dkr.ecr.ap-southeast-1.amazonaws.com`
  - `328549459982.dkr.ecr.ap-southeast-2.amazonaws.com`
  - `328549459982.dkr.ecr.ca-central-1.amazonaws.com`
  - `328549459982.dkr.ecr.eu-central-1.amazonaws.com`
  - `328549459982.dkr.ecr.eu-north-1.amazonaws.com`
  - `586180183710.dkr.ecr.eu-south-1.amazonaws.com`
  - `328549459982.dkr.ecr.eu-west-1.amazonaws.com`
  - `328549459982.dkr.ecr.eu-west-2.amazonaws.com`
  - `328549459982.dkr.ecr.eu-west-3.amazonaws.com`
  - `509306038620.dkr.ecr.me-south-1.amazonaws.com`
  - `328549459982.dkr.ecr.sa-east-1.amazonaws.com`
  - `328549459982.dkr.ecr.us-east-1.amazonaws.com`
  - `328549459982.dkr.ecr.us-east-2.amazonaws.com`
  - `328549459982.dkr.ecr.us-west-1.amazonaws.com`
  - `328549459982.dkr.ecr.us-west-2.amazonaws.com`
  - `388230364387.dkr.ecr.us-gov-east-1.amazonaws.com`
  - `347163068887.dkr.ecr.us-gov-west-1.amazonaws.com`
  - `183470599484.dkr.ecr.cn-north-1.amazonaws.com.cn`
  - `183901325759.dkr.ecr.cn-northwest-1.amazonaws.com.cn`

> 

Reviewer 3: we miss `328549459982.dkr.ecr.ap-northeast-3.amazonaws.com`

Reviewer 4: I think you also need to add:

```
386774335080.dkr.ecr.ap-southeast-3.amazonaws.com
553577323255.dkr.ecr.me-central-1.amazonaws.com
```

You may also choose to pull from regional Amazon ECR repositories such as the following.

  - `917644944286.dkr.ecr.af-south-1.amazonaws.com`
  - `375569722642.dkr.ecr.ap-east-1.amazonaws.com`
  - `328549459982.dkr.ecr.ap-northeast-1.amazonaws.com`
  - `328549459982.dkr.ecr.ap-northeast-2.amazonaws.com`
  - `328549459982.dkr.ecr.ap-south-1.amazonaws.com`
  - `328549459982.dkr.ecr.ap-southeast-1.amazonaws.com`
  - `328549459982.dkr.ecr.ap-southeast-2.amazonaws.com`
  - `328549459982.dkr.ecr.ca-central-1.amazonaws.com`
  - `328549459982.dkr.ecr.eu-central-1.amazonaws.com`
  - `328549459982.dkr.ecr.eu-north-1.amazonaws.com`
  - `586180183710.dkr.ecr.eu-south-1.amazonaws.com`
  - `328549459982.dkr.ecr.eu-west-1.amazonaws.com`
  - `328549459982.dkr.ecr.eu-west-2.amazonaws.com`
  - `328549459982.dkr.ecr.eu-west-3.amazonaws.com`
  - `509306038620.dkr.ecr.me-south-1.amazonaws.com`
  - `328549459982.dkr.ecr.sa-east-1.amazonaws.com`
  - `328549459982.dkr.ecr.us-east-1.amazonaws.com`
  - `328549459982.dkr.ecr.us-east-2.amazonaws.com`
  - `328549459982.dkr.ecr.us-west-1.amazonaws.com`
  - `328549459982.dkr.ecr.us-west-2.amazonaws.com`
  - `388230364387.dkr.ecr.us-gov-east-1.amazonaws.com`

>   - `347163068887.dkr.ecr.us-gov-west-1.amazonaws.com`

Reviewer 3: I think we miss China region here. 
```
183470599484.dkr.ecr.cn-north-1.amazonaws.com.cn/bottlerocket-update-operator
183901325759.dkr.ecr.cn-northwest-1.amazonaws.com.cn/bottlerocket-update-operator
```
